### PR TITLE
Apply the Kotlin plugin in a java project

### LINKED ISSUE #95141: Java apps might not compile if they depend on a plugin which uses old kotlin version
## Steps to Reproduce

1. Execute `flutter create app -a java` to create a java app
2. Add `fluttertoast: 8.0.8` as dependency
3. Run the app

**Expected results:** app runs

**Actual results:** compilation fails

To verify this is a problem only for java apps, do the same but without `-a java` parameter.

The fluttertoast plugin uses `1.3.50` as kotlin version. I tried manually editing to be `1.5.30` and the problem was gone.

<details>
  <summary>Logs</summary>

```
Launching lib\main.dart on Redmi Note 5 in debug mode...
e: Incompatible classes were found in dependencies. Remove them from the classpath or use '-Xskip-metadata-version-check' to suppress errors
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/095f7a2bff526ae90c824eb73c15302a/jetified-kotlinx-coroutines-android-1.5.2.jar!/META-INF/kotlinx-coroutines-android.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/325857b14eafda42f6ebd6ed8a130644/jetified-window-1.0.0-beta04-api.jar!/META-INF/window_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/35a54be3f0ff5b406b46c3399384ac55/jetified-kotlin-stdlib-jdk7-1.5.30.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/7deeafea6d161ff761d4ca76f8dcca61/jetified-kotlin-stdlib-jdk8-1.5.30.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/ae7d3805d00abe8151ad445b4c4e1688/jetified-kotlin-stdlib-common-1.5.31.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/b7eb4c7d400c07c4717df481e3ba4ffd/jetified-window-java-1.0.0-beta04-api.jar!/META-INF/window-java_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d54167437c933767a9a5f422140c1c39/jetified-kotlinx-coroutines-core-jvm-1.5.2.jar!/META-INF/kotlinx-coroutines-core.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (15, 6): Class 'kotlin.jvm.JvmStatic' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/jvm/JvmStatic.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (18, 14): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (23, 5): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (27, 5): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (31, 5): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (33, 14): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (37, 14): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\FlutterToastPlugin.kt: (38, 5): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (19, 15): Unresolved reference: Exception

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (26, 9): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (49, 17): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (52, 26): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (60, 42): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (62, 26): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (64, 30): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (67, 30): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (69, 21): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (70, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (71, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (73, 21): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (78, 42): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (81, 42): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (89, 21): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (91, 36): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (94, 36): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (97, 36): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (102, 17): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (103, 43): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (103, 66): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (105, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (107, 17): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (109, 24): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (112, 30): Unresolved reference: isInitialized
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (113, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (115, 24): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (117, 28): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.
The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (122, 22): Unresolved reference: isInitialized
e: C:\Users\danya\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\fluttertoast-8.0.8\android\src\main\kotlin\io\github\ponnamkarthik\toast\fluttertoast\MethodCallHandlerImpl.kt: (127, 21): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

The class is loaded from C:/Users/danya/.gradle/caches/transforms-2/files-2.1/d661f32da69dea35c8d36328ff95610e/jetified-kotlin-stdlib-1.5.31.jar!/kotlin/Unit.class

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':fluttertoast:compileDebugKotlin'.
> Compilation error. See log for more details
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
BUILD FAILED in 32s
[!] Your project requires a newer version of the Kotlin Gradle plugin.
    Find the latest version on https://kotlinlang.org/docs/gradle.html#plugin-and-versions, then update C:\Users\danya\Desktop\app\android\build.gradle:
    ext.kotlin_version = '<latest-version>'
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

```
[√] Flutter (Channel refresh, 2.6.0-12.0.pre.988, on Microsoft Windows [Version 10.0.19041.1348], locale ru-RU)
    • Flutter version 2.6.0-12.0.pre.988 at C:\dev\src\flutter
    • Upstream repository git@github.com:nt4f04und/flutter.git
    • FLUTTER_GIT_URL = git@github.com:nt4f04und/flutter.git
    • Framework revision 44a2c2fedc (2 days ago), 2021-12-10 19:28:53 +0300
    • Engine revision 2516ea636d
    • Dart version 2.16.0 (build 2.16.0-85.0.dev)
    • DevTools version 2.9.1

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\danya\AppData\Local\Android\sdk
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 4.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.63.0)
    • VS Code at C:\Users\danya\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.29.0

[√] Connected device (4 available)
    • Redmi Note 5 (mobile) • 40c9f14 • android-arm64  • Android 9 (API 28)
    • Windows (desktop)     • windows • windows-x64    • Microsoft Windows [Version 10.0.19041.1348]
    • Chrome (web)          • chrome  • web-javascript • Google Chrome 96.0.4664.93
    • Edge (web)            • edge    • web-javascript • Microsoft Edge 96.0.1054.34

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by blasten**
Users will need to recreate the android directory by deleting `android` and
running `fluter create .` within the app directory.

Hopefully, this won't be necessary once the auto migration tool is available. 
cc @GaryQian 

This PR also moves the Gradle, AGP, and Kotlin version values to Dart, so they are in a single place.

Fixes https://github.com/flutter/flutter/issues/95141

### CODE REVIEW

File: packages/flutter_tools/templates/plugin/android-java.tmpl/build.gradle.tmpl [hunk 1]
```diff
@@ -2,13 +2,15 @@ group '{{androidIdentifier}}'
 version '1.0'
 
 buildscript {
+    ext.kotlin_version = '1.4.32'
```
Comment (stuartmorgan) [inline packages/flutter_tools/templates/plugin/android-java.tmpl/build.gradle.tmpl:5]: I understand the app change, but why does an Android *plugin* need this?

**Review by stuartmorgan-g — CHANGES_REQUESTED** [review]
This makes copies 3 and 4 of this value in various parts of the template. We should have a test that they all match so that someone doesn't update a subset of them without remembering to update the others.

File: packages/flutter_tools/templates/plugin/android-java.tmpl/build.gradle.tmpl (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/templates/plugin/android-java.tmpl/build.gradle.tmpl]: This isn't needed. done

**Review by stuartmorgan-g — APPROVED** [review]
LGTM

**Pull request merged**